content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1601 @@
+
+
+/// type=MS, answer=[4,5]
+
+In `student.php`, which of the following are named placeholders?
+
+- `students`
+
+- `birth_date`
+
+- `min_birth_date`
+
+- `:max_birth_date`
+
+- `:min_birth_date`
+
+
+/// type=SS, answer=[1]
+
+In `student.php`, what value is associated with the named placeholder `:min_birth_date` on line 9?
+
+- `1999-01-01`
+
+- `2000-01-01`
+
+- `birth_date`
+
+- `min_birth_date`
+
+- `max_birth_date`
+
+
+/// type=SS, answer=[2]
+
+In `student.php`, what value is associated with the named placeholder `:max_birth_date` on line 9?
+
+- `1999-01-01`
+
+- `2000-01-01`
+
+- `birth_date`
+
+- `min_birth_date`
+
+- `max_birth_date`
+
+
+/// type=SS, answer=[5]
+
+Which statement best describes `WHERE birth_date >= :min_birth_date AND birth_date < :max_birth_date` on line 9 of `student.php`?
+
+- It selects the rows in the `students` table related to the values `1999-01-01` and `2000-01-01`.
+
+- It selects the rows in the `students` table equivalent to the dates `1999-01-01` and `2000-01-01`.
+
+- It returns all records in the `students` table whose `birth_date` starts with `1999-01-01` or `2000-01-01`.
+
+- It updates the rows in the `students` table whose `birth_date` is between the dates `1999-01-01` and `2000-01-01`.
+
+- It filters the rows in the `students` table whose `birth_date` is greater than or equal to the date `1999-01-01` and less than the date `2000-01-01`.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe `$pstmt = $conn->prepare($sql)` on line 11 of `student.php`?
+
+- It executes the SQL statement.
+
+- It prepares the argument `$sql` for execution.
+
+- It returns the result set as a `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, which of the following represents the parameter name in the statement `$pstmt->bindParam(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR);` on line 12?
+
+- `$data`
+
+- `PDO::PARAM_STR`
+
+- `:min_birth_date`
+
+- `$data['min_birth_date']`
+
+- `(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR)`
+
+
+/// type=SS, answer=[4]
+
+In `student.php`, which of the following represents the value to bind in the SQL statement on line 12?
+
+- `$data`
+
+- `PDO::PARAM_STR`
+
+- `:min_birth_date`
+
+- `$data['min_birth_date']`
+
+- `(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR)`
+
+
+/// type=SS, answer=[3]
+
+In `student.php`, which argument represents the explicit data type in the statement `$pstmt->bindParam(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR);` on line 12?
+
+- `$data`
+
+- `min_birth_date`
+
+- `PDO::PARAM_STR`
+
+- `:min_birth_date`
+
+- `$data['min_birth_date']`
+
+
+/// type=MS, answer=[1,4]
+
+In `student.php`, which statements correctly describe `(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR)` on line 12?
```

Comment:
Is it possible to remove the parentheses `()` from the question here? Please check. Thanks.